### DISCUSSION TITLE
Enhanced media transcoding

### DISCUSSION THREAD

**Discussion by pljones** [music-assistant/support#2576] [Feature requests and ideas]
Hi,

I've recently discovered CATT (https://github.com/skorokithakis/catt) and had started writing a web-based media service to allow Google Nest Hub devices to play media file from the local LAN.

Then my husband pointed me to Music Assistant Server and it looked like it could save a lot of time as it has support for both ChromeCast and local and SMB file storage.

The major stumbling blocks, unfortunately, have been:
- recognising media file types -- such as mkv
- transcoding media file types for playback based on the supported media types of the rendering device

CATT provides much wider support - based on ffmpeg, I believe.

With both Music Assistant Server and CATT written in python, I was wondering if there could be any synergistic development between the two projects that would overcome the above (and possibly other) current limitations in Music Assistant Server and speed its development.

Thanks,

-- Peter

**Comment by OzGav** [comment]
MA uses ffmpeg

**Comment by marcelveldt** [comment]
MKV is a container format for video files, Music Assistant is for audio/music only

**Reply by Rudd-O** [reply]
MKV many times contains Opus audio, others AAC.  I have many of these files (songs).  Can MA list and play them?  It sure can with MP4 files.  FFMPEG can also play audio from MKV.  Why artificially limit these songs from being added to the database?

**Reply by pljones** [reply]
Exactly.  Currently I have to rip the audio from a live concert just to I can hear it.  That doesn't make sense: storing the same audio twice.

**Reply by Rudd-O** [reply]
Fascinating — I found myself in the exact same predicament.

**Reply by marcelveldt** [reply]
If the usecase is to only hear audio from a container format then that is perfectly doable but it needs an optional toggle because some people deliberately want the scanner to ignore video files. Hence our current whitelisting on extension

**Reply by pljones** [reply]
Sure, I can understand some people not wanting a four hour concert appearing in the middle of a set of random tracks from their library.  But that's more a shortcoming of not being able to organise the library content flexibly than needing to enforce filtering by file type.

**Reply by Rudd-O** [reply]
@marcelveldt I would be perfectly fine with a toggle somewhere in the local media provider.  I definitely want to *not ignore* video files — my music folder contains video tracks as well as audio tracks, but it doesn't contain feature length films.

**Comment by pljones** [comment]
OK, so neither of the above addresses the core point of improvements being possible to Music Assistant through investigating what CATT could offer, only objections.

Maybe some constructive feedback could be attempted?

**Comment by OzGav** [comment]
> Cast All The Things allows you to send videos from many, many online sources (YouTube, Vimeo, and a few hundred others) to your Chromecast

So that is video, MA is audio? MA already supports local file storage and streams to cast devices (as well a a good number of others). You mention in passing the "current limitations in Music Assistant Server" yet don't say what you think those are. Again MA's purpose is to stream high quality audio to audio devices.

**Reply by pljones** [reply]
One example, MA has a hard coded list of file extensions, rather than checking the content of the file for the contained media type.  Doing that would be a good start -- basically, let ffmpeg/ffprobe do the hard work, rather than trying to code in constraints that aren't really needed.

**Reply by OzGav** [reply]
That has nothing to do with CATT

**Reply by pljones** [reply]
MA blocks certain file types that contain playable audio.

Why?

Is it a political statement?  Religious believe?  Or is there a conspiracy going on to prevent this audio being hear?

We should be told...

**Reply by OzGav** [reply]
I am assuming you are trying to be humourous. You are commenting on a topic that is two years old and has no votes. This is not on our radar at all.

**Reply by Rudd-O** [reply]
I just voted it up.  Please add support for playback of music videos — **only audio playback** is **fine**.
